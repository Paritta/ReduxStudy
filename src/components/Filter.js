import React from "react";
import styled from "styled-components";
import oc from "open-color";
import FaApple from "react-icons/lib/fa/apple";
import FaFeed from "react-icons/lib/fa/feed";
import FaFlask from "react-icons/lib/fa/flask";
import FaHeartbeat from "react-icons/lib/fa/heartbeat";
import FaRecycle from "react-icons/lib/fa/recycle";

const Wrapper = styled.div`
    text-align: left;
    margin: 50px 0;
`;

const FilterBox = styled.li`
    list-style: none;
`;

const FilterItem = styled.div`
    width: 150px;
    height: 150px;
    
    border-radius: 100%;
    background: ${oc.gray[3]};
    
    &:hover {
        background: ${oc.gray[2]};
    }
    
    transition: background 1s ease;
    
    display: flex;
    justify-content: center;
    align-items: center;
    
    cursor: pointer;
`;

const FilterUl = styled.ul`
    margin: 0;
    padding: 0;
    
    display: flex;
    flex-direction: row;
    justify-content: space-around;
`;

const Filter = ({ filterMaterial, filterNursing, filterInformation, filterComputer, filterChemical }) => {
    return (
        <div>
            <Wrapper>
                <FilterUl>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                                filterMaterial("filter_Material")
                            }}>
                            <FaRecycle size={70} color="white"/>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterNursing("filter_Nursing")
                        }}>
                            <FaHeartbeat size={70} color="white"/>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                                filterInformation("filter_Information")
                            }}>
                            <FaApple size={70} color="white"/>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterComputer("filter_Computer")
                        }}>
                            <FaFeed size={70} color="white"/>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterChemical("filter_Chemical")
                        }}>
                            <FaFlask size={70} color="white"/>
                        </FilterItem>
                    </FilterBox>
                </FilterUl>
            </Wrapper>
        </div>
    )
};

export default Filter;