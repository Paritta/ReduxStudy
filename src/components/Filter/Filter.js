import React from "react";
import styled from "styled-components";
import oc from "open-color";
import FaApple from "react-icons/lib/fa/apple";
import FaFeed from "react-icons/lib/fa/feed";
import FaFlask from "react-icons/lib/fa/flask";
import FaHeartbeat from "react-icons/lib/fa/heartbeat";
import FaRecycle from "react-icons/lib/fa/recycle";

const Wrapper = styled.div`
    text-align: left;
    background: ${oc.gray[1]};
`;

const FilterBox = styled.li`
    list-style: none;
`;

const FilterItem = styled.div`
    width: 250px;
    height: 250px;
    
    border-radius: 100%;
    background: white;
    color: ${oc.gray[6]};
    
    &:hover {
        transform: translateY(-15px);
    }
    
    transition: transform 1s ease;
    
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    
    cursor: pointer;
`;

const FilterUl = styled.ul`
    margin: 0;
    padding: 0;
    
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    height: 50vh;
`;

const Span = styled.span`
    padding-top: 15px;
    text-size: 1.5em;
`;

export const Filter = ({ filterMaterial, filterNursing, filterInformation, filterComputer, filterChemical }) => {
    return (
        <div>
            <Wrapper>
                <FilterUl>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                                filterMaterial("filter_Material")
                            }}>
                            <FaRecycle size={100} color="black"/>
                            <Span>
                                신소재 공학과
                            </Span>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterNursing("filter_Nursing")
                        }}>
                            <FaHeartbeat size={100} color="black"/>
                            <Span>
                                간호학과
                            </Span>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                                filterInformation("filter_Information")
                            }}>
                            <FaApple size={100} color="black"/>
                            <Span>
                                정보통신학과
                            </Span>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterComputer("filter_Computer")
                        }}>
                            <FaFeed size={100} color="black"/>
                            <Span>
                                컴퓨터공학과
                                </Span>
                        </FilterItem>
                    </FilterBox>
                    <FilterBox>
                        <FilterItem
                            onClick={() => {
                            filterChemical("filter_Chemical")
                        }}>
                            <FaFlask size={100} color="black"/>
                            <Span>
                                화학공학과
                            </Span>
                        </FilterItem>
                    </FilterBox>
                </FilterUl>
            </Wrapper>
        </div>
    )
};

export default Filter;