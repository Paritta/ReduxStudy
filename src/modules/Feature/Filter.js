import { createAction, handleActions } from "redux-actions";

const FILTER_MATERIAL = "filter/filter_Material";
const FILTER_NURSING = "filter/filter_Nursing";
const FILTER_INFORMATION = "filter/filter_Information";
const FILTER_COMPUTER = "filter/filter_Computer";
const FILTER_CHEMICAL = "filter/filter_Chemical";

export const filterMaterial = createAction(FILTER_MATERIAL);
export const filterNursing = createAction(FILTER_NURSING);
export const filterInformation = createAction(FILTER_INFORMATION);
export const filterComputer = createAction(FILTER_COMPUTER);
export const filterChemical = createAction(FILTER_CHEMICAL);

const initialState = {
    Filter: "filter_Material"
};

const reducer = handleActions({
    [FILTER_MATERIAL]: (state, action) => {
        return {
            Filter: action.payload
        }
    },
    [FILTER_NURSING]: (state, action) => {
        return {
            Filter: action.payload
        }
    },
    [FILTER_INFORMATION]: (state, action) => {
        return {
            Filter: action.payload
        }
    },
    [FILTER_COMPUTER]: (state, action) => {
        return {
            Filter: action.payload
        }
    },
    [FILTER_CHEMICAL]: (state, action) => {
        return {
            Filter: action.payload
        }
    },
}, initialState);

export default reducer;
