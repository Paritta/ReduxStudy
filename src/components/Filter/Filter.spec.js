import React from "react";
import { Filter } from "./Filter";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Card render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Filter />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
