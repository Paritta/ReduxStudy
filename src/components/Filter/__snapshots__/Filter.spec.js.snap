// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render correctly 1`] = `
.c0 {
  text-align: left;
  background: #f1f3f5;
}

.c1 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50vh;
}

.c2 {
  list-style: none;
}

.c3 {
  width: 250px;
  height: 250px;
  border-radius: 100%;
  background: white;
  color: #868e96;
  -webkit-transition: -webkit-transform 1s ease;
  -webkit-transition: transform 1s ease;
  transition: transform 1s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c3:hover {
  -webkit-transform: translateY(-15px);
  -ms-transform: translateY(-15px);
  transform: translateY(-15px);
}

.c4 {
  padding-top: 15px;
  text-size: 1.5em;
}

<div>
  <div
    className="c0"
  >
    <ul
      className="c1"
    >
      <li
        className="c2"
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={100}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "black",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={100}
          >
            <g>
              <path
                d="m18.7 26.1l-0.4 8.2 0 0.5-9.4-0.6q-0.8-0.1-1.5-0.8t-1-1.4q-0.3-0.6-0.4-1.2t0.1-1.5 0.3-1.2 0.5-1.5 0.4-1.1q1.7 0.2 11.4 0.6z m-8.7-13.1l4 8.5-3.2-2.1q-1.4 1.6-2.5 3.2t-1.6 2.8-0.9 2.1-0.4 1.4l-0.1 0.5-4.3-8q-0.4-0.5-0.4-1.2t0.2-1.1l0.1-0.4q0.8-1.4 2.6-4.1l-3.1-2z m27.5 11.6l-4.2 8q-0.3 0.6-0.8 1t-1 0.5l-0.4 0.1q-1.6 0.1-4.9 0.2l0.2 3.7-5.1-8.2 4.7-8.1 0.1 3.9q3.8 0.3 6.4 0.1t3.7-0.8z m-17.5-20.7q-1.1 1.4-5.9 9.7l-7.1-4.1-0.4-0.3 5-7.9q0.4-0.7 1.3-1.1t1.8-0.2q0.5 0.1 1.1 0.3t0.9 0.5 1 0.7 0.8 0.8 0.8 0.8 0.7 0.8z m14.6 6.9l4.7 8.1q0.4 0.8 0.3 1.7t-0.6 1.6q-0.3 0.5-0.7 0.9t-0.9 0.6-1.1 0.5-1 0.3-1.2 0.3-1 0.3q-0.8-1.6-5.9-9.7l7-4.4z m-3.2-5.1l3.2-1.8-4.9 8.3-9.4-0.4 3.4-2q-0.8-1.9-1.7-3.7t-1.7-2.7-1.4-1.8-1.1-1l-0.3-0.3 9 0q0.7-0.1 1.3 0.2t0.9 0.7l0.2 0.3q0.9 1.4 2.5 4.2z"
              />
            </g>
          </svg>
          <span
            className="c4"
          >
            신소재 공학과
          </span>
        </div>
      </li>
      <li
        className="c2"
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={100}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "black",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={100}
          >
            <g>
              <path
                d="m28.6 22.9h6.8q-0.1 0.1-0.2 0.2t-0.2 0.2l-0.1 0-13.9 13.4q-0.4 0.4-1 0.4t-1-0.4l-13.9-13.4q-0.1 0-0.5-0.4h8.3q0.4 0 0.8-0.3t0.5-0.8l1.6-6.3 4.2 14.9q0.2 0.4 0.6 0.7t0.8 0.3q0.5 0 0.9-0.3t0.5-0.7l3.2-10.8 1.3 2.5q0.4 0.8 1.3 0.8z m11.4-9.6q0 3.2-2.3 6.7h-8.2l-2.5-4.9q-0.2-0.4-0.6-0.6t-0.8-0.2q-1 0.1-1.2 1l-2.9 9.6-4.4-15.3q-0.1-0.4-0.5-0.7t-0.9-0.3-0.9 0.3-0.5 0.7l-2.6 10.4h-9.4q-2.3-3.5-2.3-6.7 0-4.9 2.8-7.7t7.9-2.7q1.4 0 2.8 0.4t2.7 1.3 2.1 1.6 1.7 1.5q0.8-0.8 1.7-1.5t2.1-1.6 2.7-1.3 2.8-0.4q5 0 7.9 2.7t2.8 7.7z"
              />
            </g>
          </svg>
          <span
            className="c4"
          >
            간호학과
          </span>
        </div>
      </li>
      <li
        className="c2"
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={100}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "black",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={100}
          >
            <g>
              <path
                d="m35.6 27.1q-0.9 2.8-2.8 5.6-2.8 4.4-5.7 4.4-1.1 0-3.1-0.7-1.9-0.8-3.4-0.8-1.3 0-3.2 0.8-1.8 0.7-2.9 0.7-3.4 0-6.7-5.7-3.3-5.9-3.3-11.3 0-5.1 2.5-8.3 2.5-3.2 6.4-3.2 1.6 0 3.9 0.6 2.3 0.7 3.1 0.7 1 0 3.2-0.7 2.3-0.8 3.8-0.8 2.7 0 4.8 1.4 1.2 0.8 2.3 2.3-1.7 1.5-2.5 2.6-1.5 2.1-1.5 4.6 0 2.8 1.6 5t3.5 2.8z m-8.4-26.2q0 1.4-0.6 3.1-0.7 1.7-2.1 3.1-1.2 1.2-2.4 1.6-0.9 0.2-2.4 0.3 0.1-3.3 1.8-5.7 1.6-2.4 5.6-3.3 0 0.1 0 0.2t0.1 0.3q0 0.1 0 0.2t0 0.2z"
              />
            </g>
          </svg>
          <span
            className="c4"
          >
            정보통신학과
          </span>
        </div>
      </li>
      <li
        className="c2"
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={100}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "black",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={100}
          >
            <g>
              <path
                d="m13.1 30q0 1.8-1.3 3t-3 1.3-3.1-1.3-1.2-3 1.3-3 3-1.3 3 1.3 1.3 3z m11.4 2.7q0 0.7-0.4 1.1-0.4 0.5-1 0.5h-3q-0.6 0-1-0.4t-0.4-0.9q-0.5-5.1-4.2-8.7t-8.7-4.2q-0.6 0-0.9-0.4t-0.4-1v-3q0-0.6 0.5-1 0.3-0.4 0.9-0.4h0.1q3.6 0.3 6.9 1.8t5.8 4q2.5 2.6 4 5.8t1.8 6.8z m11.4 0.1q0.1 0.6-0.4 1-0.4 0.5-1 0.5h-3.2q-0.6 0-1-0.4t-0.4-1q-0.3-4.8-2.3-9.1t-5.1-7.5-7.5-5.1-9.2-2.3q-0.5 0-0.9-0.5t-0.4-0.9v-3.2q0-0.6 0.4-1 0.5-0.4 1-0.4h0.1q5.8 0.2 11.2 2.6t9.5 6.6q4.2 4.1 6.6 9.5t2.6 11.2z"
              />
            </g>
          </svg>
          <span
            className="c4"
          >
            컴퓨터공학과
          </span>
        </div>
      </li>
      <li
        className="c2"
      >
        <div
          className="c3"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={100}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": "black",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width={100}
          >
            <g>
              <path
                d="m35.6 32.3q1.2 2 0.5 3.4t-3.2 1.4h-25.7q-2.4 0-3.1-1.4t0.5-3.4l11.2-17.7v-8.9h-1.4q-0.6 0-1-0.4t-0.5-1 0.5-1 1-0.4h11.4q0.6 0 1 0.4t0.4 1-0.4 1-1 0.4h-1.4v8.9z m-17.4-16.2l-6.1 9.6h15.9l-6.1-9.6-0.4-0.7v-9.7h-2.9v9.7z"
              />
            </g>
          </svg>
          <span
            className="c4"
          >
            화학공학과
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
